tests/bootstrap4_badge/test_plugins.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+from cms.api import add_plugin
+from cms.test_utils.testcases import CMSTestCase
+
+from djangocms_bootstrap4.contrib.bootstrap4_badge.cms_plugins import (
+    Bootstrap4BadgePlugin,
+)
+
+from ..fixtures import B4TestFixture
+
+
+class B4BadgePluginTestCase(B4TestFixture, CMSTestCase):
+
+    def test_plugin(self):
+        plugin = add_plugin(
+            placeholder=self.placeholder,
+            plugin_type=Bootstrap4BadgePlugin.__name__,
+            language=self.language,
+            badge_text="some text",
+        )
+        plugin.full_clean()
+        self.page.publish(self.language)
+
+        with self.login_user_context(self.superuser):
+            response = self.client.get(self.request_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, '<span class="badge badge-primary">')
+        self.assertContains(response, 'some text')
```

Comment:
the problem on most is, even when adding html, it still throws errors sometimes:

```AssertionError: False is not true : Couldn't find '<span class="badge badge-primary">' in response```

Reply:
Yes, because in HTML mode, it searches for entire subtrees of the document, so you cannot just search for the opening tag of an element if it has any other contents. That's why I suggested searching for the entire string including `some test` and the closing tag. However, if the element contains anything else beyond that, which you don't want to hard-code in the test, then it's a no-go – I'm not familiar with the output here, so it's hard for me to tell.